src/core/server/saved_objects/import/lib/check_origin_conflicts.ts

```diff
@@ -85,12 +97,27 @@ const checkOriginConflict = async (
     namespace,
     objectIdsBeingImported,
     pendingOverwrites,
+    retryMap,
+    retryDestinations,
   } = params;
   const { type, originId, id } = object;
 
-  if (!typeRegistry.isMultiNamespace(type) || pendingOverwrites.has(`${type}:${id}`)) {
+  const key = getObjectKey(object);
+  const retry = retryMap.get(key);
+  if (
+    !typeRegistry.isMultiNamespace(type) ||
+    pendingOverwrites.has(key) ||
+    !!retry?.destinationId
+  ) {
     // Skip the search request for non-multi-namespace types, since by definition they cannot have inexact matches or ambiguous conflicts.
     // Also skip the search request for objects that we've already determined have an "exact match" conflict.
+    // Finally, skip the search request for objects that have specified a destinationId for a retry.
```

Comment:
All these code comments are fantastic! Do we have a dev guide that explains the intricacies of sharing saved objects? If not, we could consider adding these to one (but leave them in the code too 😄 )

Reply:
Thank you!

Our dev guide for sharing saved objects is here: https://www.elastic.co/guide/en/kibana/8.0/sharing-saved-objects.html